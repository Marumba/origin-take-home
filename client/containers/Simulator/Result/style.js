import styled from '@emotion/styled';

import * as S from '../shared/style.shared';

export const Result = styled(S.ContentHolder)`
	background: #fff;

	display: flex;
	align-items: center;

	${({ customCss, theme }) => (typeof customCss === 'function' ? customCss(theme) : customCss)}
`;

export const Title = styled.div`
	${({ theme }) => theme.mixins.subtitle}
	text-align: left;
	flex: 1;
`;
export const ResultValue = styled.div`
	${({ theme }) => theme.mixins.headingMedium}
	text-align: right;
	flex: 1;
`;
