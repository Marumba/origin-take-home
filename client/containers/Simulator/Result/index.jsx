import React from 'react';
import PropTypes from 'prop-types';

import * as S from './style';

function Result({ title, customCss }) {
	return (
		<S.Result data-testid="simulator-result" customCss={customCss}>
			{title && <S.Title>{title}</S.Title>}
			<S.ResultValue className="highEmphasis">123213</S.ResultValue>
		</S.Result>
	);
}

Result.propTypes = {
	customCss: PropTypes.oneOfType([PropTypes.func, PropTypes.shape({ styles: PropTypes.string })]),
	title: PropTypes.string
};

Result.defaultProps = {
	customCss: undefined,
	title: undefined
};

export default Result;
