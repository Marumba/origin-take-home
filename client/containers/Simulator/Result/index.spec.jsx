import React from 'react';

import { render, screen } from '~/helpers/testing.customRtl';

import Result from '.';

describe('Container Simulator Result', () => {
	it('should render without any props given', () => {
		const { container } = render(<Result />);
		expect(screen.getByTestId('simulator-result')).toBeInTheDocument();

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should display the texts from given title props', () => {
		render(<Result title="Title" />);
		expect(screen.getByText('Title')).toBeInTheDocument();
	});

	it('should incorporate de given css', () => {
		render(
			<Result
				customCss={{
					background: 'red'
				}}
			/>
		);
		expect(screen.getByTestId('simulator-result')).toHaveStyleRule('background', 'red');
	});
});
