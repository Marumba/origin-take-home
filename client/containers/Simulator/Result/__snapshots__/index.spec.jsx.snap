// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Simulator Result should render without any props given 1`] = `
.emotion-0 {
  padding: 24px;
  background: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  font-family: 'Rubik',sans-serif;
  font-weight: 500;
  font-size: 24px;
  line-height: 120%;
  text-align: right;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-2.highEmphasis {
  color: #0079FF;
}

<div
  class="emotion-0 emotion-1"
  data-testid="simulator-result"
>
  <div
    class="highEmphasis emotion-2 emotion-3"
  >
    123213
  </div>
</div>
`;
